providers/os/resources/parse.go

```diff
@@ -130,6 +132,190 @@ func (s *mqlParseJson) params(content string) (interface{}, error) {
 	return res, nil
 }
 
+func initParseXml(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if err := fileFromPathOrContent(runtime, args); err != nil {
+		return nil, nil, err
+	}
+
+	return args, nil, nil
+}
+
+func (s *mqlParseXml) id() (string, error) {
+	if s.File.Data == nil {
+		return "", errors.New("no file provided for parse.json")
+	}
+
+	file := s.File.Data
+	return file.Path.Data, nil
+}
+
+func (s *mqlParseXml) content(file *mqlFile) (string, error) {
+	c := file.GetContent()
+	return c.Data, c.Error
+}
+
+type xmlElem struct {
+	attributes map[string]string
+	children   []*xmlElem
+	data       string
+	isElement  bool
+}
+
+func (x *xmlElem) addAttr(a []xml.Attr) {
+	if len(a) == 0 {
+		return
+	}
+	x.attributes = map[string]string{}
```

Comment:
This always re-initializes the map.
```suggestion
	if x.attributes == nil {
		x.attributes = map[string]string{}
	}
```

Reply:
Thank you @afiune